nixos/common/detect-hostname-change.nix

```diff
@@ -0,0 +1,48 @@
+# Protection against deploying system closures to the wrong host.
+{
+  config,
+  lib,
+  ...
+}:
+{
+  options.srvos.detect-hostname-change = {
+    enable = lib.mkEnableOption "warn if the hostname changes between deploys" // {
+      default = true;
+    };
+  };
+
+  config = lib.mkIf (config.srvos.detect-hostname-change.enable && config.networking.hostName != "") {
+    system.preSwitchChecks.detectHostnameChange = ''
+      # Ignore if the system is getting installed
+      if [[ ! -e /run/current-system ]]; then
+        exit
+      fi
+
+      actual=$(< /proc/sys/kernel/hostname)
+      desired=${config.networking.hostName}
+      if [[ "$actual" = "$desired" ]]; then
+        exit
+      fi
+
+      # Useful for automation
+      if [[ "''${EXPECTED_HOSTNAME:-}" = "$desired" ]];
```

Comment:
@zimbatm : This line is missing a "then", which makes bash exit with a parsing error 2 lines below.

I thought I wrote this comment somewhere on GitHub about 2 minutes ago, but can't find it anymore. Better ping twice than never I guess. ;-)

Reply:
https://github.com/nix-community/srvos/pull/595